node/actors/network/src/gossip/batch_votes.rs

```diff
@@ -57,13 +72,14 @@ impl BatchVotes {
     /// It exits as soon as an invalid entry is found.
     /// `self` might get modified even if an error is returned
     /// (all entries verified so far are added).
+    ///
     /// Returns true iff some new entry was added.
     pub(super) fn update(
         &mut self,
         attesters: &attester::Committee,
         data: &[Arc<attester::Signed<attester::Batch>>],
-    ) -> anyhow::Result<bool> {
-        let mut changed = false;
+    ) -> anyhow::Result<BatchUpdateStats> {
+        let mut stats = BatchUpdateStats::default();
```

Comment:
why not report these stats directly?

Reply:
I made these [tests](https://github.com/matter-labs/era-consensus/blob/99df3002508d65f49ebb829b9a104dee4e577ea6/node/actors/network/src/gossip/tests/mod.rs#L611) to look for quorum and `BatchVotes` is easy to test in isolation. Adding global metrics into it feels disturbing. It even gets cloned IIRC by the subscribers. 

By contrast there is only one instance of `BatchVotesWatch` and it's meant to coordinate changes to the state by managing locks and subscriptions, so it felt like a better place to maintain the metrics. 

The [vise](https://crates.io/crates/vise) page recommends two ways to make metrics testable:
> * You may pass around references to the metric type(s) in the logic under test so that these types can be injected and then checked by tests.
> * Alternatively, your logic may produce statistics that are then reported as metrics (this may be beneficial for performance as well). In this case, produced statistics can be checked by tests

Because I didn't want to actually test that metrics are emitted, the stats approach seemed cleaner.

Reply:
ok